src/libraries/System.Formats.Tar/tests/TarTestsBase.Gnu.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using Xunit;
+
+namespace System.Formats.Tar.Tests
+{
+    public abstract partial class TarTestsBase : FileCleanupTestBase
+    {
+        protected void SetRegularFile(GnuTarEntry regularFile)
+        {
+            SetCommonRegularFile(regularFile);
+            SetPosixProperties(regularFile);
+            SetGnuProperties(regularFile);
+        }
+
+        protected void SetDirectory(GnuTarEntry directory)
+        {
+            SetCommonDirectory(directory);
+            SetPosixProperties(directory);
+            SetGnuProperties(directory);
+        }
+
+        protected void SetHardLink(GnuTarEntry hardLink)
+        {
+            SetCommonHardLink(hardLink);
+            SetPosixProperties(hardLink);
+            SetGnuProperties(hardLink);
+        }
+
+        protected void SetSymbolicLink(GnuTarEntry symbolicLink)
+        {
+            SetCommonSymbolicLink(symbolicLink);
+            SetPosixProperties(symbolicLink);
+            SetGnuProperties(symbolicLink);
+        }
+
+        protected void SetCharacterDevice(GnuTarEntry characterDevice)
+        {
+            SetCharacterDeviceProperties(characterDevice);
+            SetGnuProperties(characterDevice);
+        }
+
+        protected void SetBlockDevice(GnuTarEntry blockDevice)
+        {
+            SetBlockDeviceProperties(blockDevice);
+            SetGnuProperties(blockDevice);
+        }
+
+        protected void SetFifo(GnuTarEntry fifo)
+        {
+            SetFifoProperties(fifo);
+            SetGnuProperties(fifo);
+        }
+
+        protected void SetGnuProperties(GnuTarEntry entry)
+        {
+            DateTimeOffset approxNow = DateTimeOffset.Now.Subtract(TimeSpan.FromMinutes(2));
```

Comment:
I am guessing these tests will fail when we enter and leave daylight savings. Not sure whether that's easily solved, it probably shouldn't block committing this.

Reply:
Good point. Need to think of a way to get this properly verified without having it fail on DST.

Reply:
I improved most of the timestamp verifications with a predefined datetime that I modify and set in my entries.